pkg/speaker/bgp/config/bgp_configs.go

```diff
@@ -0,0 +1,6416 @@
+			} else if !r.Equal(l) {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+// struct for container rpol:defined-sets.
+// Predefined sets of attributes used in policy match
+// statements.
+type DefinedSets struct {
+	// original -> rpol:prefix-sets
+	// Enclosing container for defined prefix sets for matching.
+	PrefixSets []PrefixSet `mapstructure:"prefix-sets" json:"prefix-sets,omitempty"`
+	// original -> rpol:neighbor-sets
+	// Enclosing container for defined neighbor sets for matching.
+	NeighborSets []NeighborSet `mapstructure:"neighbor-sets" json:"neighbor-sets,omitempty"`
+	// original -> rpol:tag-sets
+	// Enclosing container for defined tag sets for matching.
+	TagSets []TagSet `mapstructure:"tag-sets" json:"tag-sets,omitempty"`
+	// original -> bgp-pol:bgp-defined-sets
+	// BGP-related set definitions for policy match conditions.
+	BgpDefinedSets BgpDefinedSets `mapstructure:"bgp-defined-sets" json:"bgp-defined-sets,omitempty"`
+}
+
+func (lhs *DefinedSets) Equal(rhs *DefinedSets) bool {
+	if lhs == nil || rhs == nil {
+		return false
+	}
+	if len(lhs.PrefixSets) != len(rhs.PrefixSets) {
+		return false
+	}
+	{
+		lmap := make(map[string]*PrefixSet)
+		for i, l := range lhs.PrefixSets {
+			lmap[mapkey(i, string(l.PrefixSetName))] = &lhs.PrefixSets[i]
+		}
+		for i, r := range rhs.PrefixSets {
+			if l, y := lmap[mapkey(i, string(r.PrefixSetName))]; !y {
+				return false
+			} else if !r.Equal(l) {
+				return false
+			}
+		}
+	}
+	if len(lhs.NeighborSets) != len(rhs.NeighborSets) {
+		return false
+	}
+	{
+		lmap := make(map[string]*NeighborSet)
+		for i, l := range lhs.NeighborSets {
+			lmap[mapkey(i, string(l.NeighborSetName))] = &lhs.NeighborSets[i]
+		}
+		for i, r := range rhs.NeighborSets {
+			if l, y := lmap[mapkey(i, string(r.NeighborSetName))]; !y {
+				return false
+			} else if !r.Equal(l) {
+				return false
+			}
+		}
+	}
+	if len(lhs.TagSets) != len(rhs.TagSets) {
+		return false
+	}
+	{
+		lmap := make(map[string]*TagSet)
+		for i, l := range lhs.TagSets {
+			lmap[mapkey(i, string(l.TagSetName))] = &lhs.TagSets[i]
+		}
+		for i, r := range rhs.TagSets {
+			if l, y := lmap[mapkey(i, string(r.TagSetName))]; !y {
+				return false
+			} else if !r.Equal(l) {
+				return false
+			}
+		}
+	}
+	if !lhs.BgpDefinedSets.Equal(&(rhs.BgpDefinedSets)) {
+		return false
+	}
+	return true
+}
+
+// struct for container rpol:routing-policy.
+// top-level container for all routing policy configuration.
+type RoutingPolicy struct {
+	// original -> rpol:defined-sets
+	// Predefined sets of attributes used in policy match
+	// statements.
+	DefinedSets DefinedSets `mapstructure:"defined-sets" json:"defined-sets,omitempty"`
+	// original -> rpol:policy-definitions
+	// Enclosing container for the list of top-level policy
+	// definitions.
+	PolicyDefinitions []PolicyDefinition `mapstructure:"policy-definitions" json:"policy-definitions,omitempty"`
+}
+
+func (lhs *RoutingPolicy) Equal(rhs *RoutingPolicy) bool {
```

Comment:
Do we use this `Equal` method?

Reply:
No, we don't. `bgp_configs.go` is autogenerated from an [OpenConfig Yang model](https://github.com/osrg/gobgp/blob/master/tools/pyang_plugins/gobgp.yang) as done in [GoBGP](https://github.com/osrg/gobgp/blob/42a5f1d771d574c0759701444246e8e1b9585efc/internal/pkg/config/bgp_configs.go). I assume there might be a way to outline the spec to produce only the required output. It may be suitable to deal with that in another PR.